src/storage/txn/scheduler.rs

```diff
@@ -978,6 +979,28 @@ impl<E: Engine, L: LockManager> Scheduler<E, L> {
         resumable_wake_up_list
     }
 
+    fn on_acquired_locks_finished(&self, new_acquired_locks: Vec<kvrpcpb::LockInfo>) {
```

Comment:
As it increases execution costs compared with the previous implementation, it might still impact performance if the server load is high. Maybe we could run such verifications in the benchbot cases.